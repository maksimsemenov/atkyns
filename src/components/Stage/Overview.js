import React, { PropTypes } from 'react'
import Selector from 'components/controls/Selector/Selector'
import Field from 'components/controls/Field'
import ButtonLink from 'components/controls/ButtonLink/ButtonLink'
import l from 'utils/local'
import { STAGE_PETITIONER } from 'constants/stages'
import { RELASHIONSHIP, P_STATUS_TYPE, R_USV_CURRENT_STATUS } from 'constants/dataFieldNames'

const pStatusOptions = [
  { value: 'citizen', name: l('US Citizen') },
  { value: 'resident', name: l('Permanent resident') }
]
const relationshipOptions = [
  { value: 'spouse', name: l('Spouse') },
  { value: 'parent', name: l('Parent') },
  { value: 'child', name: l('Child') },
  // { value: 'daughter', name: l('Daughter') },
  { value: 'brother', name: l('Brother') },
  { value: 'sister', name: l('Sister') },
]
const rUSStatusOprions = [
  { value: true, name: l('Yes') },
  { value: false, name: l('No') },
]

const Overview = ({ caseId }) => {
  return (
    <div className='stage'>
      <div className='stage__field'>
        <div className='stage__fieldTitle'>{l('Your current status')}</div>
        <div className='stage__fieldDescription'></div>
        <div className='stage__fieldControl'>
          <Field
            fieldName={P_STATUS_TYPE}
            caseId={caseId}
            component={Selector}
            options={pStatusOptions}
          />
        </div>
      </div>
      <div className='stage__field'>
        <div className='stage__fieldTitle'>{l('Relationship')}</div>
        <div className='stage__fieldControl'>
          <Field
            fieldName={RELASHIONSHIP}
            caseId={caseId}
            component={Selector}
            options={relationshipOptions}
          />
        </div>
      </div>
      <div className='stage__field'>
        <div className='stage__fieldTitle'>{l('Is you relative now in United States?')}</div>
        <div className='stage__fieldControl'>
          <Field
            fieldName={R_USV_CURRENT_STATUS}
            caseId={caseId}
            component={Selector}
            options={rUSStatusOprions}
          />
        </div>
      </div>
      <div className='stage__controls'>
        <ButtonLink to={`/case/${caseId}/${STAGE_PETITIONER}`}>Continue</ButtonLink>
      </div>
    </div>
  )
}

Overview.propTypes = {
  caseId: PropTypes.string.isRequired
}


export default Overview
