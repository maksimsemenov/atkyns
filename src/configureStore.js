import { createStore, applyMiddleware, compose } from 'redux'
import { fromJS } from 'immutable'
import throttle from 'lodash/throttle'
import thunk from 'redux-thunk'
import { loadState, saveState } from 'localStorage'
import reducer from 'reducers'

let middleware = [thunk]

const composeEnhancers =
  typeof window === 'object' &&
  window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ ?
    window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({}) :
    compose

const enhancer = composeEnhancers(applyMiddleware(...middleware))

const configureStore = () => {
  const persistedState = fromJS({ cases: loadState() })
  const store = createStore(
    reducer,
    persistedState,
    enhancer
  )

  store.subscribe(throttle(() => {
    saveState(store.getState().get('cases').toJS())
  }), 2000)

  return store
}
export default configureStore
