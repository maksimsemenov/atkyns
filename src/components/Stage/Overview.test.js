import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import Router from 'react-router-dom/BrowserRouter'
import configureStore from 'configureStore'
import Overview from 'components/Stage/Overview'

describe('Overview component', () => {
  it('Snapshot test', () => {
    const store = configureStore()
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Overview caseId='32sdasdljk' />
          </Router>
        </Provider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
