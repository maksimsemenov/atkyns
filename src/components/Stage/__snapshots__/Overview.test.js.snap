exports[`Overview component Snapshot test 1`] = `
<div
  className="stage">
  <div
    className="stage__field">
    <div
      className="stage__fieldTitle">
      Your current status
    </div>
    <div
      className="stage__fieldDescription" />
    <div
      className="stage__fieldControl">
      <div
        className="selector__control">
        <ul
          className="selector has-value">
          <li
            className="selector__option"
            onClick={[Function onClick]}>
            US Citizen
          </li>
          <li
            className="selector__option"
            onClick={[Function onClick]}>
            Permanent resident
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    className="stage__field">
    <div
      className="stage__fieldTitle">
      Relationship
    </div>
    <div
      className="stage__fieldControl">
      <div
        className="selector__control">
        <ul
          className="selector has-value">
          <li
            className="selector__option"
            onClick={[Function onClick]}>
            Spouse
          </li>
          <li
            className="selector__option"
            onClick={[Function onClick]}>
            Parent
          </li>
          <li
            className="selector__option"
            onClick={[Function onClick]}>
            Child
          </li>
          <li
            className="selector__option"
            onClick={[Function onClick]}>
            Brother
          </li>
          <li
            className="selector__option"
            onClick={[Function onClick]}>
            Sister
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    className="stage__field">
    <div
      className="stage__fieldTitle">
      Is you relative now in United States?
    </div>
    <div
      className="stage__fieldControl">
      <div
        className="selector__control">
        <ul
          className="selector has-value">
          <li
            className="selector__option"
            onClick={[Function onClick]}>
            Yes
          </li>
          <li
            className="selector__option"
            onClick={[Function onClick]}>
            No
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    className="stage__controls">
    <a
      className="button"
      href="/case/32sdasdljk/petitioner"
      onClick={[Function anonymous]}>
      Continue
    </a>
  </div>
</div>
`;
